samples/bluetooth/peripheral_hids_mouse/src/main.c

```diff
@@ -162,14 +166,23 @@ static void advertising_continue(void)
 
 	if (!k_msgq_get(&bonds_queue, &addr, K_NO_WAIT)) {
 		char addr_buf[BT_ADDR_LE_STR_LEN];
+		int err;
+
+		if (atomic_test_bit(&adv_state, ADV_RUNNING)) {
```

Comment:
I would say that we do not need atomic here. Every access to adv_state seems to be done from the same no preemptive context

Reply:
`adv_state` is used in the connected and disconnected callbacks. As far as I know, these callbacks are called from different threads, BT and sysworkq.

Reply:
Yes and they are cooperative threads. You also call advertising_continue only from sysworkq

Reply:
Right. Changed to bool.